raft/raft.go

```diff
@@ -823,6 +823,11 @@ func stepLeader(r *raft, m pb.Message) {
 		return
 	case pb.MsgReadIndex:
 		if r.quorum() > 1 {
+			if r.raftLog.zeroTermOnErrCompacted(r.raftLog.term(r.raftLog.committed)) != r.Term {
+				// Reject read only request when this leader has not committed any log entry at its term.
```

Comment:
It's usual to drop message like this. E.g. Requests like proposes are usually dropped in raft. And it's possible the a MsgReadIndex is lost due to network issue when follower forwards it to leader.
There should be a timeout/retry of requests outside this module.